### PR TITLE
Feature/45 dto booking

### LINKED ISSUE #64: Implement ResponseHandler in Booking

### PR DESCRIPTION
**Pull request by royschrauwen**
Implemented a BookingDTO and did some small changes on tests and the implementation of the ResponseHandler in Offer and Booking.

This should also close #64 and #65

### CODE REVIEW

File: src/main/java/avd/inf/jdm/rentmycar/controller/OfferController.java [hunk 1]
```diff
@@ -28,52 +28,50 @@ public OfferController(OfferService offerService, CarService carService) {
...
+            return ResponseHandler.generateResponse("Offer could not be created", HttpStatus.BAD_REQUEST, null);
 
-        return ResponseHandler.generateResponse("Offer could not be created", HttpStatus.BAD_REQUEST, null);
+        } catch (IllegalArgumentException e) {
+            return ResponseHandler.generateResponse(e.getMessage(), HttpStatus.BAD_REQUEST, null);
         }
+    }
 
 
     @PutMapping("/v1/offers/{id}")
```
Comment (robfuncken) [inline src/main/java/avd/inf/jdm/rentmycar/controller/OfferController.java:77]: Im not sure if we should implement the DTO for PUT. What do you think?

**Review by robfuncken — COMMENTED** [review]
Looks good Roy! I've only one question, otherwise looks good for me to merge.

File: src/main/java/avd/inf/jdm/rentmycar/controller/OfferController.java (hunk 1 above)
Comment (royschrauwen) [reply src/main/java/avd/inf/jdm/rentmycar/controller/OfferController.java:77]: I am not sure. I don't think it is needed?

**Comment by royschrauwen** [comment]
> Looks good Roy! I've only one question, otherwise looks good for me to merge.

Thanks for the review, Rob!
I will merge this and I suggest we pick up the DTO-question in Teams. 👍

**Pull request merged**